src/UKHO.ERPFacade.API/Helpers/EncContentSapMessageBuilder.cs

```diff
@@ -339,12 +349,53 @@ private static XmlElement BuildAction(XmlDocument soapXml, Product product, Unit
                 }
                 actionAttributeList.Add((node.SortingOrder, itemSubNode));
             }
+
+            foreach (var node in action.Attributes.Where(x => x.Section == UkhoWeekNumberSection))
+            {
+                XmlElement itemSubNode = soapXml.CreateElement(node.XmlNodeName);
+
+                if (node.IsRequired)
+                {
+                    if (IsValidWeekNumber(ukhoWeekNumber))
+                    {
+                        switch (node.XmlNodeName)
+                        {
+                            case ValidFrom:
+                                string thursdayDate = s_weekDetailsProvider.GetThursdayDateOfWeek(
+                                ukhoWeekNumber.Year, ukhoWeekNumber.Week);
+                                itemSubNode.InnerText = GetXmlNodeValue(thursdayDate);
+                                break;
+
+                            case WeekNo:
+                                string weekData = GetUkhoWeekNumberData(ukhoWeekNumber);
+                                itemSubNode.InnerText = GetXmlNodeValue(weekData);
+                                break;
+
+                            case Correction:
+                                itemSubNode.InnerText = ukhoWeekNumber.CurrentWeekAlphaCorrection ? GetXmlNodeValue("Y") : GetXmlNodeValue("N");
```

Comment:
Can we add constant variable for both Correction values

Reply:
Resolved